### PR TITLE
Add Net datatype

### PR DESCRIPTION
**Pull request by martenson**
as defined at https://genome.ucsc.edu/goldenPath/help/net.html

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/config/sample/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -989,6 +989,7 @@
     <!-- Flexible Image Transport System (FITS) used in Astronomy https://fits.gsfc.nasa.gov/ https://fits.gsfc.nasa.gov/rfc4047.txt -->
     <datatype extension="fits" type="galaxy.datatypes.binary:FITS" mimetype="application/octet-stream" display_in_upload="true" description="Flexible Image Transport System (FITS) used in Astronomy"/>
     <datatype extension="chain" type="galaxy.datatypes.chain:Chain" display_in_upload="true"/>
+    <datatype extension="net" type="galaxy.datatypes.chain:Net" display_in_upload="true"/>
```
Comment (bgruening) [inline lib/galaxy/config/sample/datatypes_conf.xml.sample]: add the link as url or description as well to the upstream doc?

Comment (mvdbeek) [reply lib/galaxy/config/sample/datatypes_conf.xml.sample]: Should we use something like `ucsc.net` as the extension ? `net` appears to be used in other applications: https://file.org/extension/net

Comment (martenson) [reply lib/galaxy/config/sample/datatypes_conf.xml.sample]: thanks! resolved in d4aa9c4

**Comment by mvdbeek** [comment]
New datatypes can also go to a stable release, feel free to open a PR against 24.0 if that would be helpful @martenson

**Pull request merged**